pkg/workloadmeta/testing/store.go

```diff
@@ -75,23 +75,39 @@ func (s *Store) GetKubernetesPod(id string) (*workloadmeta.KubernetesPod, error)
 }
 
 // GetKubernetesPodForContainer returns a KubernetesPod that contains the
-// specified containerID.
+// specified containerID. The ownership information might lag because the kubelet check sets the owner field
+// but a container might be stored by CRI checks. Thus, the function might return an error when the pod
+// actually exists.
 func (s *Store) GetKubernetesPodForContainer(containerID string) (*workloadmeta.KubernetesPod, error) {
-	entities, ok := s.store[workloadmeta.KindKubernetesPod]
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	containerEntities, ok := s.store[workloadmeta.KindContainer]
 	if !ok {
 		return nil, errors.NewNotFound(containerID)
 	}
 
-	for _, e := range entities {
-		pod := e.(*workloadmeta.KubernetesPod)
-		for _, podContainer := range pod.Containers {
-			if podContainer.ID == containerID {
-				return pod, nil
-			}
-		}
+	containerEntity, ok := containerEntities[containerID]
+	if !ok {
+		return nil, errors.NewNotFound(containerID)
+	}
+
+	container := containerEntity.(*workloadmeta.Container)
+	if container.Owner == nil || container.Owner.Kind != workloadmeta.KindKubernetesPod {
+		return nil, errors.NewNotFound(containerID)
+	}
+
+	podEntities, ok := s.store[workloadmeta.KindKubernetesPod]
```

Comment:
This can be replace by `s.GetKubernetesPod(container.Owner.ID)`

Reply:
It should be fixed by now